rust/agama-lib/src/scripts/model.rs

```diff
@@ -48,39 +45,21 @@ pub enum ScriptsGroup {
 pub struct BaseScript {
     pub name: String,
     #[serde(flatten)]
-    pub source: ScriptSource,
+    pub source: FileSource,
 }
 
 impl BaseScript {
+    /// Writes the script to the given directory.
+    ///
+    /// * `workdir`: directory to write the script to.
     fn write<P: AsRef<Path>>(&self, workdir: P) -> Result<(), ScriptError> {
         let script_path = workdir.as_ref().join(&self.name);
         std::fs::create_dir_all(script_path.parent().unwrap())?;
-
-        let mut file = fs::OpenOptions::new()
-            .create(true)
-            .write(true)
-            .truncate(true)
-            .mode(0o500)
-            .open(&script_path)?;
-
-        match &self.source {
-            ScriptSource::Text { content } => write!(file, "{}", &content)?,
-            ScriptSource::Remote { url } => Transfer::get(url, &mut file)?,
-        };
-
+        self.source.write(&script_path, 0o500)?;
```

Comment:
Not writable?
For the `auth login` file we also omitted the write bit and it did bite us.